primer-io/primer-sdk-ios
# [FEATURE][DEX-1044] Introduce New Translations

DEX-1044

# What this PR does
## Translator Script
Added the translator script that is mainly responsible for changing the translations by fetching the latest translations from Phrase.

## New Languages 
This PR introduces the support for the languages listed below:
- Danish
- Dutch
- German
- Greek
- Italian
- Norwegian
- Polish
- Portuguese
- Spanish
- Swedish
- Turkish

# Notable decisions & other stuff
Decided to include the script as a git submodule to use it for Android as well.

# How to Review
Unfortunately we cannot see a proper diff for the `Localizable.strings` files due to the lack of support for `utf-16` encoding in git.

In order to tackle this and see a proper diff on your local machine by converting `utf-16` to `utf-8`:
Add the following to your `~/.gitconfig` file
```
[diff "utf16"]
textconv = "iconv -f utf-16 -t utf-8"
```

Create or edit your `~/.gitattributes` file and add the following
```
*.strings diff=utf16
```



# Instructions on how to test this
- Run the Example app on a simulator
- Set the system language to `Greek`, `Turkish`, or a newly introduced language
- Initialize Primer SDK
- Check if the `back button` label on top-left is translated
- Click on `Universal Checkout`
- You should see the new translations on a similar screen below
<img width="561" alt="Screen Shot 2022-02-03 at 11 19 00" src="https://user-images.githubusercontent.com/95745429/152305800-14fe7c49-be4f-461a-87bf-26d080c2ffee.png">

<img width="561" alt="Screen Shot 2022-02-02 at 10 56 13" src="https://user-images.githubusercontent.com/95745429/152114863-254c4e7a-bae4-4aa0-9777-3cb924f9f6ef.png">



# Before merging

_QA_

- [ ] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk
